api4/post_test.go

```diff
@@ -326,18 +328,338 @@ func TestCreatePost(t *testing.T) {
 		CheckForbiddenStatus(t, resp)
 	})
 
-	// TODO: need some way of hosting and running a plugin in this context
-	// t.Run("create post against topic", func(t *testing.T) {
-	// 	topicType := "topic_type"
-	// 	topicId := model.NewId()
+	t.Run("no permissions on registered topic", func(t *testing.T) {
```

Comment:
Can we split this test into two, one with the plugin "not registering UserHasPermissionToCollection" and one where it's explicitly returning false?